# Первые 10 строк каждого CSV-датасета

В этом ноутбуке каждый датасет открыт в отдельной ячейке. Используется `pandas.read_csv(..., nrows=10)` и затем `.head(10)`, чтобы не загружать огромные CSV целиком в память.

`numpy` тоже импортирован, но метод `head()` относится к pandas, не к NumPy.

In [2]:
from pathlib import Path

import numpy as np
import pandas as pd

pd.set_option("display.max_columns", 80)
pd.set_option("display.max_rows", 20)
pd.set_option("display.max_colwidth", 80)

PROJECT_ROOT = Path.cwd()
if not (PROJECT_ROOT / "data").exists():
    PROJECT_ROOT = PROJECT_ROOT.parent

DATA_DIR = PROJECT_ROOT / "data"
DATA_DIR

WindowsPath('c:/Users/MSI/PycharmProjects/hseml-group-project-bezdari_3k/data')

In [3]:
def show_head(file_name: str, n_rows: int = 10) -> pd.DataFrame:
    path = DATA_DIR / file_name
    df = pd.read_csv(path, nrows=n_rows)
    print(f"{file_name}: первые {len(df)} строк, показано {df.shape[1]} колонок")
    return df.head(n_rows)

## `correlations.csv`

**Что это за датасет:** корреляционная матрица для полной Featuretools-матрицы.

**За что отвечает:** показывает, насколько признаки линейно связаны друг с другом. Значения лежат примерно от `-1` до `1`: ближе к `1` - сильная положительная связь, ближе к `-1` - сильная отрицательная связь, около `0` - слабая линейная связь.

**Как читать колонки:** названия колонок - это признаки из исходной feature matrix. Каждая строка тоже соответствует признаку, но в этом файле имя строки не вынесено отдельной колонкой, поэтому первые значения выглядят как корреляции признаков между собой.

**Для чего используем:** для EDA, поиска признаков, связанных с `TARGET`, и поиска сильно похожих признаков. Для обучения моделей этот файл не используем.

In [4]:
show_head("correlations.csv")

correlations.csv: первые 10 строк, показано 1985 колонок


,SK_ID_CURR,REG_REGION_NOT_LIVE_REGION,EXT_SOURCE_2,FLAG_PHONE,AMT_INCOME_TOTAL,AMT_CREDIT,REGION_RATING_CLIENT,REG_REGION_NOT_WORK_REGION,CNT_FAM_MEMBERS,DAYS_BIRTH,REG_CITY_NOT_LIVE_CITY,COMMONAREA_AVG,FLAG_WORK_PHONE,FLAG_DOCUMENT_21,ELEVATORS_AVG,LIVINGAPARTMENTS_MODE,FLOORSMAX_MODE,FLAG_DOCUMENT_17,AMT_REQ_CREDIT_BUREAU_QRT,COMMONAREA_MEDI,FLAG_EMAIL,FLAG_DOCUMENT_5,DAYS_REGISTRATION,DAYS_ID_PUBLISH,LANDAREA_MODE,NONLIVINGAPARTMENTS_MEDI,FLAG_DOCUMENT_15,REGION_POPULATION_RELATIVE,LIVE_REGION_NOT_WORK_REGION,LIVINGAREA_MEDI,FLOORSMIN_MODE,ELEVATORS_MEDI,AMT_REQ_CREDIT_BUREAU_YEAR,FLAG_EMP_PHONE,FLAG_DOCUMENT_13,LIVINGAPARTMENTS_MEDI,DEF_30_CNT_SOCIAL_CIRCLE,EXT_SOURCE_3,FLAG_DOCUMENT_8,LANDAREA_MEDI,...,MODE(credit.NAME_CONTRACT_STATUS)_Completed,MODE(credit.NAME_CONTRACT_STATUS)_Signed,MODE(bureau.CREDIT_TYPE)_Another type of loan,MODE(bureau.CREDIT_TYPE)_Car loan,MODE(bureau.CREDIT_TYPE)_Consumer credit,MODE(bureau.CREDIT_TYPE)_Credit card,MODE(bureau.CREDIT_TYPE)_Loan for business development,MODE(bureau.CREDIT_TYPE)_Loan for the purchase of equipment,MODE(bureau.CREDIT_TYPE)_Loan for working capital replenishment,MODE(bureau.CREDIT_TYPE)_Microloan,MODE(bureau.CREDIT_TYPE)_Mortgage,MODE(bureau.CREDIT_TYPE)_Real estate loan,MODE(bureau.CREDIT_TYPE)_Unknown type of loan,MODE(previous_app.NAME_SELLER_INDUSTRY)_Auto technology,MODE(previous_app.NAME_SELLER_INDUSTRY)_Clothing,MODE(previous_app.NAME_SELLER_INDUSTRY)_Connectivity,MODE(previous_app.NAME_SELLER_INDUSTRY)_Construction,MODE(previous_app.NAME_SELLER_INDUSTRY)_Consumer electronics,MODE(previous_app.NAME_SELLER_INDUSTRY)_Furniture,MODE(previous_app.NAME_SELLER_INDUSTRY)_Industry,MODE(previous_app.NAME_SELLER_INDUSTRY)_Jewelry,MODE(previous_app.NAME_SELLER_INDUSTRY)_MLM partners,MODE(previous_app.NAME_SELLER_INDUSTRY)_Tourism,MODE(previous_app.NAME_SELLER_INDUSTRY)_XNA,MODE(previous_app.MODE(credit.NAME_CONTRACT_STATUS))_Active,MODE(previous_app.MODE(credit.NAME_CONTRACT_STATUS))_Completed,MODE(previous_app.MODE(credit.NAME_CONTRACT_STATUS))_Signed,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Active,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Approved,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Completed,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Demand,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Returned to the store,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Signed,MODE(bureau.MODE(bureau_balance.STATUS))_0,MODE(bureau.MODE(bureau_balance.STATUS))_1,MODE(bureau.MODE(bureau_balance.STATUS))_2,MODE(bureau.MODE(bureau_balance.STATUS))_3,MODE(bureau.MODE(bureau_balance.STATUS))_5,MODE(bureau.MODE(bureau_balance.STATUS))_C,MODE(bureau.MODE(bureau_balance.STATUS))_X
0,1.000000,-0.000283,0.002342,0.002753,-0.001820,-0.000343,-0.001075,0.001097,-0.002895,-0.001500,-0.001885,-0.001463,-0.000415,0.000282,0.004862,0.003589,0.004386,0.001450,0.001025,-0.001036,0.000281,-0.001097,-0.000973,-0.000384,0.001548,-0.003020,0.002604,0.000849,0.002903,0.002205,0.002081,0.005067,0.004659,-0.001337,0.000896,0.003272,-0.000082,0.000222,0.001809,0.001699,...,-0.000947,0.000079,0.000332,-0.001994,-0.001309,0.000938,-0.001190,0.000442,0.001010,-0.000414,0.001183,-0.002528,-0.002080,-0.002346,-0.000115,0.001993,0.000715,-0.001417,-0.001020,0.000296,0.001745,-0.002253,-0.002141,0.000822,0.002259,-0.000788,0.000079,0.001526,0.001403,0.001954,-0.001378,-0.004109,0.000725,0.000571,0.000592,-0.001078,-0.001853,-0.002603,-0.000913,-0.000626
1,-0.000283,1.000000,0.015570,0.002078,0.031191,0.024010,-0.044166,0.450804,-0.017133,0.065486,0.339232,0.015360,0.064484,0.005006,0.014890,0.006482,0.009304,0.004498,-0.001813,0.015031,0.018632,0.009517,0.028213,0.034757,-0.014236,0.001464,-0.002017,0.002118,0.088263,0.005179,0.012279,0.014290,-0.017095,0.036640,0.003407,0.009547,-0.007940,-0.049162,0.023669,-0.013746,...,0.000183,-0.000625,0.003379,0.008322,-0.036635,0.033207,-0.001962,-0.000316,-0.000806,-0.000152,0.001206,-0.000224,-0.000592,-0.000978,0.001998,0.00

## `correlations_spec.csv`

**Что это за датасет:** корреляционная матрица для компактного `spec`-набора признаков.

**За что отвечает:** делает то же самое, что `correlations.csv`, но для меньшего набора признаков. Это удобнее для анализа, потому что колонок меньше.

**Как читать колонки:** колонка `Unnamed: 0` содержит название признака-строки. Остальные колонки - признаки, с которыми считается корреляция. Например, значение на пересечении строки `EXT_SOURCE_3` и колонки `TARGET` показывает связь `EXT_SOURCE_3` с дефолтом.

**Для чего используем:** для EDA и объяснения зависимостей в compact/spec-наборе. Для обучения моделей напрямую не используем.

In [5]:
show_head("correlations_spec.csv")

correlations_spec.csv: первые 10 строк, показано 1173 колонок


,Unnamed: 0,SK_ID_CURR,FLAG_EMP_PHONE,DEF_60_CNT_SOCIAL_CIRCLE,LIVE_REGION_NOT_WORK_REGION,YEARS_BEGINEXPLUATATION_AVG,APARTMENTS_MODE,FLAG_MOBIL,FLOORSMIN_MEDI,BASEMENTAREA_AVG,LANDAREA_MODE,FLAG_DOCUMENT_21,AMT_GOODS_PRICE,COMMONAREA_MODE,APARTMENTS_MEDI,EXT_SOURCE_1,EXT_SOURCE_3,FLAG_DOCUMENT_16,LIVINGAPARTMENTS_MODE,ENTRANCES_MEDI,YEARS_BEGINEXPLUATATION_MODE,APARTMENTS_AVG,FLAG_DOCUMENT_12,FLAG_DOCUMENT_5,DAYS_LAST_PHONE_CHANGE,ELEVATORS_MODE,BASEMENTAREA_MODE,FLAG_DOCUMENT_13,FLAG_DOCUMENT_3,REGION_RATING_CLIENT,FLAG_DOCUMENT_6,AMT_REQ_CREDIT_BUREAU_DAY,COMMONAREA_AVG,DAYS_BIRTH,FLAG_DOCUMENT_14,FLOORSMAX_MEDI,OBS_30_CNT_SOCIAL_CIRCLE,YEARS_BEGINEXPLUATATION_MEDI,CNT_FAM_MEMBERS,FLOORSMAX_MODE,...,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Buying a used car,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Car repairs,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Education,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Everyday expenses,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Furniture,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Gasification / water supply,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Hobby,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Journey,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Medicine,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Money for a third person,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Other,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Payments on other loans,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Purchase of electronic equipment,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Refusal to name the goal,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Repairs,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Urgent needs,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_Wedding / gift / holiday,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_XAP,MODE(previous_app.NAME_CASH_LOAN_PURPOSE)_XNA,MODE(previous_app.NAME_YIELD_GROUP)_XNA,MODE(previous_app.NAME_YIELD_GROUP)_high,MODE(previous_app.NAME_YIELD_GROUP)_low_action,MODE(previous_app.NAME_YIELD_GROUP)_low_normal,MODE(previous_app.NAME_YIELD_GROUP)_middle,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Active,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Approved,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Completed,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Demand,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Returned to the store,MODE(previous_app.MODE(cash.NAME_CONTRACT_STATUS))_Signed,MODE(bureau.MODE(bureau_balance.STATUS))_0,MODE(bureau.MODE(bureau_balance.STATUS))_1,MODE(bureau.MODE(bureau_balance.STATUS))_2,MODE(bureau.MODE(bureau_balance.STATUS))_3,MODE(bureau.MODE(bureau_balance.STATUS))_5,MODE(bureau.MODE(bureau_balance.STATUS))_C,MODE(bureau.MODE(bureau_balance.STATUS))_X,MODE(previous_app.MODE(credit.NAME_CONTRACT_STATUS))_Active,MODE(previous_app.MODE(credit.NAME_CONTRACT_STATUS))_Completed,MODE(previous_app.MODE(credit.NAME_CONTRACT_STATUS))_Signed
0,SK_ID_CURR,1.000000,-0.001337,0.001187,0.002903,0.001551,0.001961,0.002804,0.002837,-0.002070,0.001548,0.000282,-0.000232,-0.001058,0.001988,0.000082,0.000222,-0.000724,0.003589,-0.002583,0.001890,0.001556,-0.001045,-0.001097,-0.000858,0.005017,-0.001411,0.000896,-0.003411,-0.001075,0.002121,-0.002193,-0.001463,-0.001500,-0.001077,0.004588,-0.001409,0.001366,-0.002895,0.004386,...,0.000065,-0.000578,-0.001423,0.000233,0.001845,0.002266,-0.003523,0.002701,-0.000347,0.000106,-0.000259,-0.000675,-0.001571,-0.001716,0.000190,-0.001796,-0.000157,-0.000045,0.001182,0.002193,-0.000010,-0.000898,0.000820,-0.002122,0.001526,0.001403,0.001954,-0.001378,-0.004109,0.000725,0.000571,0.000592,-0.001078,-0.001853,-0.002603,-0.000913,-0.000626,0.002259,-0.000788,0.000079
1,FLAG_EMP_PHONE,-0.001337,1.000000,-0.014034,0.096447,-0.008672,0.014720,-0.000845,0.016486,0.001292,0.008657,0.008579,0.063487,0.013112,0.016323,-0.294147,-0.115293,0.043008,0.019736,-0.004350,-0.008676,0.016441,0.001195,0.019778,-0.021103,0.008927,0.000138,0.025878,0.248534,-0.032871,-0.597732,-0.000438,0.012612,0.619888,0.023559,0.016121,-0.005732,-0.008454,0

## `feature_importances.csv`

**Что это за датасет:** таблица важностей признаков для основной матрицы `feature_matrix.csv`.

**За что отвечает:** показывает, какие признаки оказались наиболее полезными в модели автора датасета. Чем выше `importance`, тем сильнее признак помогал модели.

**Как читать колонки:** `index` - исходный индекс признака, `feature` - название признака, `importance` - численная важность признака.

**Для чего используем:** для отбора top-N признаков, интерпретации и объяснения, какие признаки важны для кредитного риска. Сам по себе файл не содержит клиентов и `TARGET`, поэтому для обучения как X/y он не подходит.

In [6]:
show_head("feature_importances.csv")

feature_importances.csv: первые 10 строк, показано 3 колонок


,index,feature,importance
0,24,ORGANIZATION_TYPE,5504
1,74,EXT_SOURCE_1,2304
2,2,EXT_SOURCE_2,2298
3,42,EXT_SOURCE_3,2298
4,105,AMT_ANNUITY,1121
5,10,DAYS_BIRTH,1108
6,6,AMT_CREDIT,1015
7,524,MAX(bureau.DAYS_CREDIT),906
8,27,DAYS_ID_PUBLISH,727
9,95,DAYS_EMPLOYED,723


## `feature_matrix.csv`

**Что это за датасет:** основная Featuretools-признаковая матрица.

**За что отвечает:** хранит данные для машинного обучения: каждая строка - отдельный клиент/кредитная заявка, каждая колонка - признак клиента или агрегированный признак из связанных таблиц Home Credit.

**Как читать ключевые колонки:** `SK_ID_CURR` - идентификатор клиента/заявки, `TARGET` - целевая переменная, `0` означает нет дефолта, `1` означает дефолт, `-999` означает Kaggle test row без разметки. Остальные колонки - признаки: доход, сумма кредита, регион, возраст, телефонные флаги, агрегаты по прошлым кредитам и т.д.

**Для чего используем:** можно использовать для обучения моделей, но файл большой. В нашем CP1 вместо него выбран более лёгкий `feature_matrix_spec.csv`.

In [7]:
show_head("feature_matrix.csv")

feature_matrix.csv: первые 10 строк, показано 1698 колонок


,SK_ID_CURR,NAME_TYPE_SUITE,REG_REGION_NOT_LIVE_REGION,EXT_SOURCE_2,OCCUPATION_TYPE,FLAG_PHONE,AMT_INCOME_TOTAL,AMT_CREDIT,REGION_RATING_CLIENT,REG_REGION_NOT_WORK_REGION,CNT_FAM_MEMBERS,DAYS_BIRTH,REG_CITY_NOT_LIVE_CITY,COMMONAREA_AVG,FLAG_WORK_PHONE,HOUSETYPE_MODE,FLAG_DOCUMENT_21,ELEVATORS_AVG,LIVINGAPARTMENTS_MODE,set,FLOORSMAX_MODE,FLAG_DOCUMENT_17,AMT_REQ_CREDIT_BUREAU_QRT,COMMONAREA_MEDI,FLAG_EMAIL,FLAG_DOCUMENT_5,ORGANIZATION_TYPE,FLAG_OWN_REALTY,DAYS_REGISTRATION,DAYS_ID_PUBLISH,LANDAREA_MODE,NONLIVINGAPARTMENTS_MEDI,FLAG_DOCUMENT_15,REGION_POPULATION_RELATIVE,LIVE_REGION_NOT_WORK_REGION,LIVINGAREA_MEDI,FLOORSMIN_MODE,ELEVATORS_MEDI,NAME_INCOME_TYPE,AMT_REQ_CREDIT_BUREAU_YEAR,...,MAX(bureau.STD(bureau_balance.MONTHS_BALANCE)),MEAN(previous_app.MAX(credit.MONTHS_BALANCE)),MIN(previous_app.SUM(credit.CNT_INSTALMENT_MATURE_CUM)),MAX(previous_app.SKEW(cash.SK_DPD_DEF)),MEAN(previous_app.MIN(installments.AMT_INSTALMENT)),SKEW(previous_app.MAX(cash.SK_DPD_DEF)),STD(previous_app.MAX(credit.MONTHS_BALANCE)),MEAN(bureau.COUNT(bureau_balance)),MAX(previous_app.SUM(credit.AMT_TOTAL_RECEIVABLE)),MIN(previous_app.SUM(credit.AMT_INST_MIN_REGULARITY)),MIN(previous_app.MEAN(credit.AMT_RECIVABLE)),SKEW(previous_app.SUM(installments.DAYS_INSTALMENT)),MAX(previous_app.MIN(credit.MONTHS_BALANCE)),MIN(previous_app.SUM(cash.CNT_INSTALMENT)),STD(previous_app.SUM(credit.SK_ID_CURR)),MEAN(previous_app.MIN(credit.AMT_TOTAL_RECEIVABLE)),MEAN(previous_app.SKEW(cash.SK_ID_CURR)),MIN(previous_app.SKEW(installments.NUM_INSTALMENT_NUMBER)),SKEW(previous_app.MAX(installments.SK_ID_CURR)),SUM(previous_app.SKEW(credit.AMT_CREDIT_LIMIT_ACTUAL)),MAX(previous_app.STD(credit.CNT_DRAWINGS_OTHER_CURRENT)),MIN(previous_app.STD(installments.AMT_INSTALMENT)),SKEW(previous_app.MAX(cash.CNT_INSTALMENT)),MEAN(previous_app.MAX(credit.CNT_INSTALMENT_MATURE_CUM)),SKEW(previous_app.STD(credit.MONTHS_BALANCE)),STD(previous_app.MAX(credit.CNT_DRAWINGS_OTHER_CURRENT)),MEAN(previous_app.MIN(credit.AMT_PAYMENT_TOTAL_CURRENT)),STD(previous_app.MIN(installments.DAYS_INSTALMENT)),SKEW(previous_app.STD(installments.AMT_PAYMENT)),MEAN(previous_app.STD(credit.AMT_RECEIVABLE_PRINCIPAL)),MIN(previous_app.MAX(credit.MONTHS_BALANCE)),MEAN(previous_app.MEAN(installments.NUM_INSTALMENT_NUMBER)),SUM(previous_app.MAX(installments.DAYS_ENTRY_PAYMENT)),MEAN(previous_app.MAX(installments.AMT_PAYMENT)),STD(previous_app.SUM(cash.CNT_INSTALMENT)),STD(previous_app.MIN(cash.CNT_INSTALMENT)),STD(previous_app.MEAN(credit.CNT_DRAWINGS_OTHER_CURRENT)),MIN(previous_app.STD(credit.CNT_DRAWINGS_OTHER_CURRENT)),MIN(previous_app.SKEW(credit.AMT_DRAWINGS_OTHER_CURRENT)),SKEW(previous_app.SUM(credit.AMT_RECEIVABLE_PRINCIPAL))
0,100001,Unaccompanied,0,0.789654,NaN,0,135000.0,568800.0,2,0,2.0,-19241,0,NaN,0,block of flats,0,NaN,NaN,test,0.1250,0,0.0,NaN,1,0,Kindergarten,Y,-5170.0,-812,NaN,NaN,0,0.018850,0,0.0514,NaN,NaN,Working,0.0,...,15.008331,NaN,NaN,0.0,3951.000000,0.0,NaN,24.571429,NaN,NaN,NaN,0.000000,NaN,20.0,NaN,NaN,0.0,0.000000,0.0,0.0,NaN,5.822679e+03,0.000000,NaN,NaN,NaN,NaN,0.000000,0.000000,NaN,NaN,2.500000,-1628.0,17397.900000,0.000000,0.000000,NaN,NaN,NaN,NaN
1,100002,Unaccompanied,0,0.262949,Laborers,1,202500.0,406597.5,2,0,1.0,-9461,0,0.0143,0,block of flats,0,0.00,0.022,train,0.0833,0,0.0,0.0144,0,0,Business Entity Type 3,Y,-3648.0,-2120,0.0377,0.0000,0,0.018801,0,0.0193,0.1250,0.00,Working,1.0,...,6.344289,NaN,NaN,0.0,9251.775000,0.0,NaN,13.750000,NaN,NaN,NaN,0.000000,NaN,456.0,NaN,NaN,0.0,0.000000,0.0,0.0,NaN,9.789775e+03,0.000000,NaN,NaN,NaN,NaN,0.000000,0.000000,NaN,NaN,10.000000,-49.0,53093.745000,0.000000,0.000000,NaN,NaN,NaN,NaN
2,100003,Family,0,0.622246,Core staff,1,270000.0,1293502.5,1,0,2.0,-16765,0,0.0605,0,block of flats,0,0.08,0.079,train,0.2917,0,0.0,0.0608,0,0,School,N,-1186.0,-291,0.0128,0.0039,0,0.003541,0,0.0558,0.3333,0.08,State servant,0.0,...,NaN,NaN,NaN,0.0,56431.860000,0.0,NaN,0.000000,NaN,NaN,NaN,-0.707094,NaN,48.0,NaN,NaN,0.0,0.000000,0.0,0.0,NaN,2.054649e+01,-0.7

## `feature_matrix_advanced.csv`

**Что это за датасет:** расширенная advanced-версия признаковой матрицы.

**За что отвечает:** содержит больше признаков, чем обычная `feature_matrix.csv`: автоматические Featuretools-агрегаты, дополнительные условия, а также ручные признаки вроде кредитных ratios.

**Как читать ключевые колонки:** `SK_ID_CURR` - идентификатор заявки, `TARGET` - целевая переменная. В этом файле test rows часто имеют `TARGET = NaN`, а не `-999`. Колонки с `NEW_...` - новые engineered-признаки, например отношение аннуитета к доходу. Колонки вида `MEAN(...)`, `MAX(...)`, `SUM(...)` - агрегаты из связанных таблиц.

**Для чего используем:** подходит для более продвинутых экспериментов, но очень широкий и тяжёлый. Для CP1 можно упомянуть как альтернативный источник признаков, но основной пайплайн лучше держать на `feature_matrix_spec.csv`.

In [8]:
show_head("feature_matrix_advanced.csv")

feature_matrix_advanced.csv: первые 10 строк, показано 3596 колонок


,SK_ID_CURR,COMMONAREA_MODE,NEW_ANNUITY_TO_INCOME_RATIO,FLAG_DOCUMENT_9,LIVINGAPARTMENTS_MODE,EXT_SOURCE_3,NAME_INCOME_TYPE,FLAG_EMP_PHONE,index,FLAG_DOCUMENT_6,BASEMENTAREA_AVG,BASEMENTAREA_MEDI,NEW_EMPLOY_TO_BIRTH_RATIO,FLOORSMIN_AVG,FLAG_DOCUMENT_16,FLAG_DOCUMENT_15,NAME_TYPE_SUITE,REGION_POPULATION_RELATIVE,NEW_SCORES_STD,FLAG_DOCUMENT_11,TOTALAREA_MODE,FLAG_DOCUMENT_20,REG_CITY_NOT_WORK_CITY,AMT_REQ_CREDIT_BUREAU_YEAR,ORGANIZATION_TYPE,FLAG_DOCUMENT_14,LANDAREA_AVG,FLOORSMAX_MEDI,ELEVATORS_MODE,ELEVATORS_AVG,DAYS_ID_PUBLISH,FLOORSMAX_MODE,REGION_RATING_CLIENT,BASEMENTAREA_MODE,FLAG_DOCUMENT_4,LIVE_REGION_NOT_WORK_REGION,NEW_CREDIT_TO_INCOME_RATIO,NEW_CAR_TO_EMPLOY_RATIO,DAYS_BIRTH,REG_REGION_NOT_WORK_REGION,...,MEAN(previous.AMT_CREDIT WHERE NAME_CONTRACT_STATUS = Unused offer),MAX(previous.MEAN(cash.index) WHERE NAME_CONTRACT_STATUS = Approved),MAX(previous.SUM(cash.index)),MAX(previous.MIN(cash.SK_DPD_DEF) WHERE NAME_CONTRACT_STATUS = Approved),MIN(bureau.DIFF(AMT_CREDIT_SUM_LIMIT by SK_ID_CURR)),MEAN(previous.MIN(installments.AMT_PAYMENT)),LAST(bureau.DIFF(DAYS_ENDDATE_FACT_ANOM by SK_ID_CURR)),SUM(previous.CUM_MEAN(HOUR_APPR_PROCESS_START by SK_ID_CURR)),MAX(previous.DAYS_DECISION_ANOM WHERE NAME_CONTRACT_STATUS = Refused),MEAN(previous.MIN(credit.CNT_DRAWINGS_CURRENT) WHERE NAME_CONTRACT_STATUS = Refused),MEAN(previous.MIN(credit.SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Unused offer),MEAN(previous.LAST(installments.NUM_INSTALMENT_NUMBER)),MAX(previous.MEAN(cash.CNT_INSTALMENT)),MAX(previous.MEAN(credit.AMT_DRAWINGS_ATM_CURRENT) WHERE NAME_CONTRACT_STATUS = Unused offer),MAX(previous.CUM_SUM(DAYS_DECISION_ANOM by SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Approved),MEAN(previous.MEAN(credit.AMT_DRAWINGS_POS_CURRENT) WHERE NAME_CONTRACT_STATUS = Canceled),MEAN(previous.MEAN(credit.CNT_INSTALMENT_MATURE_CUM) WHERE NAME_CONTRACT_STATUS = Canceled),MEAN(previous.CUM_MEAN(RATE_INTEREST_PRIMARY by SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Unused offer),MEAN(previous.MIN(credit.index)),MIN(previous.LAST(installments.AMT_PAYMENT)),MEAN(previous.SUM(installments.NUM_INSTALMENT_NUMBER)),MIN(previous.LAST(credit.AMT_PAYMENT_CURRENT)),MAX(previous.SUM(installments.NUM_INSTALMENT_VERSION) WHERE NAME_CONTRACT_STATUS = Approved),MEAN(previous.CUM_SUM(index by SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Refused),MAX(previous.CUM_MEAN(AMT_ANNUITY by SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Unused offer),MAX(previous.MIN(credit.AMT_DRAWINGS_ATM_CURRENT) WHERE NAME_CONTRACT_STATUS = Approved),MEAN(previous.MEAN(installments.index)),MEAN(previous.SUM(installments.DAYS_ENTRY_PAYMENT_ANOM) WHERE NAME_CONTRACT_STATUS = Unused offer),SUM(previous.CUM_SUM(SELLERPLACE_AREA by SK_ID_CURR)),MAX(bureau.CUM_MEAN(CREDIT_DAY_OVERDUE by SK_ID_CURR)),MEAN(previous.CUM_MEAN(RATE_DOWN_PAYMENT by SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Approved),MEAN(previous.DIFF(DAYS_FIRST_DRAWING_ANOM by SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Unused offer),MEAN(previous.MEAN(credit.CNT_DRAWINGS_POS_CURRENT) WHERE NAME_CONTRACT_STATUS = Approved),LAST(previous.MEAN(credit.AMT_RECEIVABLE_PRINCIPAL)),MAX(previous.DIFF(HOUR_APPR_PROCESS_START by SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Canceled),MEAN(bureau.CUM_MEAN(DAYS_CREDIT_UPDATE_ANOM by SK_ID_CURR)),MEAN(previous.CUM_MEAN(DAYS_LAST_DUE_ANOM by SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Refused),MEAN(previous.LAST(credit.AMT_PAYMENT_CURRENT) WHERE NAME_CONTRACT_STATUS = Canceled),MAX(previous.CUM_SUM(AMT_ANNUITY by SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Approved),MEAN(previous.CUM_SUM(CNT_PAYMENT by SK_ID_CURR) WHERE NAME_CONTRACT_STATUS = Approved)
0,100001,NaN,0.152299,0,NaN,0.159520,Working,1,0,0,0.0590,0.0590,0.121044,NaN,0,0,Unaccompanied,0.018850,0.353601,0,0.0392,0,0,0.0,Kindergarten,0,NaN,0.1250,NaN,NaN,-812,0.1250,2,0.0612,0,0,4.213333,NaN,-19241,0,...,NaN,6.136414e+06,30682072.0,0.0,0.0,3951.000000,0.0,13.000000,NaN,NaN,NaN,4.000000,4.000000,NaN,0.0,NaN,NaN,NaN,NaN,17397.900,10.000000,NaN,5.0,NaN,NaN,NaN,2.532

## `feature_matrix_article.csv`

**Что это за датасет:** признаковая матрица из статьи/демонстрационного примера автора Kaggle-датасета.

**За что отвечает:** показывает вариант подготовленных признаков, который использовался в tutorial/article-подходе. По смыслу это тоже обучающая таблица: строки - клиенты, колонки - признаки.

**Как читать ключевые колонки:** `SK_ID_CURR` - идентификатор заявки, `TARGET` - дефолт или отсутствие дефолта. Финансовые признаки вроде `AMT_ANNUITY`, `AMT_CREDIT`, `AMT_GOODS_PRICE`, `AMT_INCOME_TOTAL` описывают размер кредита, платежа, цены товара и доход клиента. Колонки `AMT_REQ_CREDIT_BUREAU_*` показывают число запросов в кредитное бюро за разные периоды.

**Для чего используем:** можно использовать как альтернативную обучающую матрицу или для сравнения с другими версиями feature matrix. В нашем CP1 это вспомогательный файл, не основной.

In [9]:
show_head("feature_matrix_article.csv")

feature_matrix_article.csv: первые 10 строк, показано 1821 колонок


,SK_ID_CURR,AMT_ANNUITY,AMT_CREDIT,AMT_GOODS_PRICE,AMT_INCOME_TOTAL,AMT_REQ_CREDIT_BUREAU_DAY,AMT_REQ_CREDIT_BUREAU_HOUR,AMT_REQ_CREDIT_BUREAU_MON,AMT_REQ_CREDIT_BUREAU_QRT,AMT_REQ_CREDIT_BUREAU_WEEK,AMT_REQ_CREDIT_BUREAU_YEAR,APARTMENTS_AVG,APARTMENTS_MEDI,APARTMENTS_MODE,BASEMENTAREA_AVG,BASEMENTAREA_MEDI,BASEMENTAREA_MODE,CNT_CHILDREN,CNT_FAM_MEMBERS,CODE_GENDER,COMMONAREA_AVG,COMMONAREA_MEDI,COMMONAREA_MODE,DAYS_BIRTH,DAYS_EMPLOYED,DAYS_ID_PUBLISH,DAYS_LAST_PHONE_CHANGE,DAYS_REGISTRATION,DEF_30_CNT_SOCIAL_CIRCLE,DEF_60_CNT_SOCIAL_CIRCLE,ELEVATORS_AVG,ELEVATORS_MEDI,ELEVATORS_MODE,EMERGENCYSTATE_MODE,ENTRANCES_AVG,ENTRANCES_MEDI,ENTRANCES_MODE,EXT_SOURCE_1,EXT_SOURCE_2,EXT_SOURCE_3,...,PERCENTILE(MEAN(credit.CNT_DRAWINGS_ATM_CURRENT)),PERCENTILE(MEAN(credit.CNT_DRAWINGS_CURRENT)),PERCENTILE(MEAN(credit.CNT_DRAWINGS_OTHER_CURRENT)),PERCENTILE(MEAN(credit.CNT_DRAWINGS_POS_CURRENT)),PERCENTILE(MEAN(credit.CNT_INSTALMENT_MATURE_CUM)),PERCENTILE(MEAN(credit.SK_DPD)),PERCENTILE(MEAN(credit.SK_DPD_DEF)),PERCENTILE(COUNT(credit)),PERCENTILE(NUM_UNIQUE(credit.NAME_CONTRACT_STATUS)),PERCENTILE(SUM(installments.NUM_INSTALMENT_VERSION)),PERCENTILE(SUM(installments.NUM_INSTALMENT_NUMBER)),PERCENTILE(SUM(installments.DAYS_INSTALMENT)),PERCENTILE(SUM(installments.DAYS_ENTRY_PAYMENT)),PERCENTILE(SUM(installments.AMT_INSTALMENT)),PERCENTILE(SUM(installments.AMT_PAYMENT)),PERCENTILE(MAX(installments.NUM_INSTALMENT_VERSION)),PERCENTILE(MAX(installments.NUM_INSTALMENT_NUMBER)),PERCENTILE(MAX(installments.DAYS_INSTALMENT)),PERCENTILE(MAX(installments.DAYS_ENTRY_PAYMENT)),PERCENTILE(MAX(installments.AMT_INSTALMENT)),PERCENTILE(MAX(installments.AMT_PAYMENT)),PERCENTILE(MIN(installments.NUM_INSTALMENT_VERSION)),PERCENTILE(MIN(installments.NUM_INSTALMENT_NUMBER)),PERCENTILE(MIN(installments.DAYS_INSTALMENT)),PERCENTILE(MIN(installments.DAYS_ENTRY_PAYMENT)),PERCENTILE(MIN(installments.AMT_INSTALMENT)),PERCENTILE(MIN(installments.AMT_PAYMENT)),PERCENTILE(MEAN(installments.NUM_INSTALMENT_VERSION)),PERCENTILE(MEAN(installments.NUM_INSTALMENT_NUMBER)),PERCENTILE(MEAN(installments.DAYS_INSTALMENT)),PERCENTILE(MEAN(installments.DAYS_ENTRY_PAYMENT)),PERCENTILE(MEAN(installments.AMT_INSTALMENT)),PERCENTILE(MEAN(installments.AMT_PAYMENT)),PERCENTILE(COUNT(installments)),PERCENTILE(SUM(bureau_balance.MONTHS_BALANCE)),PERCENTILE(MAX(bureau_balance.MONTHS_BALANCE)),PERCENTILE(MIN(bureau_balance.MONTHS_BALANCE)),PERCENTILE(MEAN(bureau_balance.MONTHS_BALANCE)),PERCENTILE(COUNT(bureau_balance)),PERCENTILE(NUM_UNIQUE(bureau_balance.STATUS))
0,124904,15808.5,325908.0,247500.0,72000.0,0.0,0.0,0.0,0.0,0.0,3.0,NaN,NaN,NaN,NaN,NaN,NaN,0.0,2.0,F,NaN,NaN,NaN,-17819.0,-3950.0,-702.0,-565.0,-7539.0,0.0,0.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.651098,0.772631,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.376351,NaN,0.523086,0.405496,0.580876,0.577748,0.343908,0.350611,0.589366,0.382186,0.375931,0.378091,0.285523,0.283438,0.586833,0.500074,0.427391,0.426944,0.454275,0.612749,0.699285,0.271150,0.563301,0.559279,0.271522,0.293566,0.495789,0.314366,0.526353,0.355410,0.208955,0.880351,0.508396
1,124907,11250.0,225000.0,225000.0,112500.0,1.0,0.0,0.0,0.0,0.0,0.0,0.0082,0.0083,0.0084,NaN,NaN,NaN,0.0,2.0,F,NaN,NaN,NaN,-22221.0,NaN,-4604.0,-222.0,-6910.0,0.0,0.0,0.00,0.00,0.0000,No,0.0345,0.0345,0.0345,NaN,0.392671,0.448962,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.376351,NaN,0.077674,0.087653,0.981084,0.975424,0.002532,0.002532,0.177018,0.121314,0.796023,0.592195,0.005362,0.005511,0.586833,0.500074,0.981978,0.977957,0.265713,0.491659,0.312556,0.141942,0.983542,0.969914,0.013107,0.016235,0.130807,NaN,NaN,NaN,NaN,0.349614,NaN
2,124908,9000.0,180000.0,180000.0,112500.0,0.0,0.0,0.0,0.0,0.0,0.0,0.0072,0.0073,0.0074,0.0000,0.0000,0.000,3.0,5.0,F,NaN,NaN,NaN,-9516.0,-2136.0,-962.0,-1284.0,-798.0,0.0,0.0,NaN,NaN,NaN,No,0.1034,0.1034,0.1034,NaN,0.533371,0.353988,...,NaN,NaN,NaN,NaN,NaN,NaN,NaN,0.376351,NaN,0.041853,0.050789,0.986670,0.984510,0.020703,0.021895,0.177018,0.063375,0.655496,0.553992,0.052130,0.051832,0.

## `feature_matrix_spec.csv`

**Что это за датасет:** компактная spec-признаковая матрица.

**За что отвечает:** это главный датасет для нашего CP1. Он содержит достаточно признаков для обучения, но при этом заметно легче, чем advanced/full версии.

**Как читать ключевые колонки:** `SK_ID_CURR` - идентификатор клиента/заявки, `TARGET` - целевая переменная. Значения `0` и `1` используются для обучения, строки с `TARGET = -999` - это Kaggle test set без известного ответа, их удаляем перед supervised learning. `ORGANIZATION_TYPE`, `NAME_INCOME_TYPE` и похожие колонки - категориальные признаки. `AMT_*`, `DAYS_*`, `EXT_SOURCE_*`, `FLAG_*` - числовые/бинарные признаки клиента и заявки.

**Для чего используем:** основной источник X/y для pipeline: очистка, split, feature engineering, baseline и все эксперименты моделей.

In [10]:
show_head("feature_matrix_spec.csv")

feature_matrix_spec.csv: первые 10 строк, показано 885 колонок


,SK_ID_CURR,FLAG_EMP_PHONE,DEF_60_CNT_SOCIAL_CIRCLE,LIVE_REGION_NOT_WORK_REGION,ORGANIZATION_TYPE,YEARS_BEGINEXPLUATATION_AVG,APARTMENTS_MODE,FLAG_MOBIL,FLOORSMIN_MEDI,BASEMENTAREA_AVG,LANDAREA_MODE,NAME_INCOME_TYPE,FLAG_DOCUMENT_21,AMT_GOODS_PRICE,COMMONAREA_MODE,APARTMENTS_MEDI,EXT_SOURCE_1,EXT_SOURCE_3,FLAG_DOCUMENT_16,WALLSMATERIAL_MODE,LIVINGAPARTMENTS_MODE,ENTRANCES_MEDI,NAME_CONTRACT_TYPE,YEARS_BEGINEXPLUATATION_MODE,APARTMENTS_AVG,FLAG_DOCUMENT_12,WEEKDAY_APPR_PROCESS_START,FLAG_DOCUMENT_5,DAYS_LAST_PHONE_CHANGE,ELEVATORS_MODE,BASEMENTAREA_MODE,FLAG_DOCUMENT_13,FLAG_DOCUMENT_3,REGION_RATING_CLIENT,FLAG_DOCUMENT_6,AMT_REQ_CREDIT_BUREAU_DAY,COMMONAREA_AVG,DAYS_BIRTH,FLAG_DOCUMENT_14,FLOORSMAX_MEDI,...,SUM(previous_app.MAX(credit.AMT_BALANCE)),SUM(previous_app.MEAN(cash.CNT_INSTALMENT)),MAX(previous_app.SUM(credit.AMT_RECEIVABLE_PRINCIPAL)),MAX(previous_app.MEAN(installments.AMT_PAYMENT)),MAX(previous_app.MIN(credit.AMT_PAYMENT_TOTAL_CURRENT)),SUM(previous_app.MEAN(credit.CNT_DRAWINGS_POS_CURRENT)),MAX(previous_app.SUM(credit.CNT_DRAWINGS_POS_CURRENT)),MIN(previous_app.MAX(cash.SK_DPD)),SUM(previous_app.MIN(cash.SK_DPD)),SUM(previous_app.MAX(credit.AMT_RECEIVABLE_PRINCIPAL)),MEAN(bureau.COUNT(bureau_balance)),SUM(previous_app.MAX(cash.SK_DPD)),MIN(previous_app.MAX(credit.MONTHS_BALANCE)),SUM(previous_app.MEAN(credit.CNT_DRAWINGS_OTHER_CURRENT)),MAX(previous_app.MIN(credit.AMT_INST_MIN_REGULARITY)),MEAN(previous_app.COUNT(installments)),SUM(previous_app.MAX(credit.AMT_PAYMENT_CURRENT)),MAX(previous_app.SUM(credit.AMT_DRAWINGS_POS_CURRENT)),MEAN(previous_app.MAX(credit.AMT_DRAWINGS_OTHER_CURRENT)),MEAN(previous_app.MAX(cash.CNT_INSTALMENT)),SUM(previous_app.MAX(installments.DAYS_INSTALMENT)),MIN(previous_app.MEAN(cash.SK_DPD)),MIN(previous_app.MAX(credit.AMT_DRAWINGS_ATM_CURRENT)),MAX(previous_app.MIN(credit.CNT_DRAWINGS_OTHER_CURRENT)),MEAN(previous_app.SUM(installments.AMT_INSTALMENT)),MEAN(previous_app.MEAN(credit.AMT_DRAWINGS_OTHER_CURRENT)),MIN(previous_app.MEAN(installments.AMT_INSTALMENT)),MEAN(previous_app.MAX(credit.SK_DPD_DEF)),SUM(previous_app.MAX(credit.CNT_DRAWINGS_POS_CURRENT)),MAX(previous_app.MEAN(installments.SK_ID_CURR)),MEAN(previous_app.MEAN(installments.DAYS_INSTALMENT)),MAX(previous_app.COUNT(credit)),MIN(previous_app.SUM(cash.MONTHS_BALANCE)),MIN(previous_app.MEAN(cash.SK_ID_CURR)),MIN(previous_app.SUM(credit.SK_DPD)),MIN(previous_app.MAX(credit.CNT_DRAWINGS_POS_CURRENT)),MIN(previous_app.SUM(credit.CNT_DRAWINGS_OTHER_CURRENT)),MAX(previous_app.SUM(cash.SK_DPD)),MAX(previous_app.MIN(cash.CNT_INSTALMENT_FUTURE)),MAX(previous_app.MEAN(credit.CNT_INSTALMENT_MATURE_CUM))
0,100001,1,0.0,0,Kindergarten,0.9732,0.0672,1,NaN,0.0590,NaN,Working,0,450000.0,NaN,0.0666,0.752614,0.159520,0,"Stone, brick",NaN,0.1379,Cash loans,0.9732,0.0660,0,TUESDAY,0,-1740.0,NaN,0.0612,0,1,2,0,0.0,NaN,-19241,0,0.1250,...,0.0,4.000000,NaN,7312.725000,NaN,0.0,NaN,0.0,0.0,0.0,24.571429,0.0,NaN,0.0,NaN,4.000000,0.0,NaN,NaN,4.0,-1619.0,0.0,NaN,NaN,29250.9000,NaN,7312.725000,NaN,0.0,100001.0,-1664.000000,0.0,-275.0,100001.0,NaN,NaN,NaN,0.0,0.0,NaN
1,100002,1,2.0,0,Business Entity Type 3,0.9722,0.0252,1,0.1250,0.0369,0.0377,Working,0,351000.0,0.0144,0.0250,0.083037,0.139376,0,"Stone, brick",0.022,0.0690,Cash loans,0.9722,0.0247,0,WEDNESDAY,0,-1134.0,0.0000,0.0383,0,1,2,0,0.0,0.0143,-9461,0,0.0833,...,0.0,24.000000,NaN,11559.247105,NaN,0.0,NaN,0.0,0.0,0.0,13.750000,0.0,NaN,0.0,NaN,19.000000,0.0,NaN,NaN,24.0,-25.0,0.0,NaN,NaN,219625.6950,NaN,11559.247105,NaN,0.0,100002.0,-295.000000,0.0,-190.0,100002.0,NaN,NaN,NaN,0.0,6.0,NaN
2,100003,1,0.0,0,School,0.9851,0.0924,1,0.3333,0.0529,0.0128,State servant,0,1129500.0,0.0497,0.0968,0.311267,NaN,0,Block,0.079,0.0345,Cash loans,0.9851,0.0959,0,MONDAY,0,-828.0,0.0806,0.0538,0,1,1,0,0.0,0.0605,-16765,0,0.2917,...,0.0,29.375000,NaN,164425.332857,NaN,0.0,NaN,0.0,0.0,0.0,0.000000,0.0,NaN,0.0,NaN,8.333333,0.0,NaN,NaN,10.0,-3163.0,0.0,NaN,NaN,539621.5500,NaN,6731.115000,NaN,0.0,100003.0,-1164.333333,0.0,-

## `fi_fma.csv`

**Что это за датасет:** таблица важностей признаков для advanced feature matrix.

**За что отвечает:** показывает, какие признаки из расширенной матрицы `feature_matrix_advanced.csv` были наиболее важны для модели автора/эксперимента.

**Как читать колонки:** `Unnamed: 0` - порядковый индекс строки, `feature` - название признака, `importance` - важность. Большие значения `importance` означают, что признак чаще/сильнее использовался моделью.

**Для чего используем:** для анализа advanced-признаков и возможного отбора top-N признаков, если решим строить модель на `feature_matrix_advanced.csv`. Для текущего CP1 это справочный файл.

In [11]:
show_head("fi_fma.csv")

fi_fma.csv: первые 10 строк, показано 3 колонок


,Unnamed: 0,feature,importance
0,0,COMMONAREA_MODE,37.6
1,1,NEW_ANNUITY_TO_INCOME_RATIO,192.4
2,2,FLAG_DOCUMENT_9,0.0
3,3,LIVINGAPARTMENTS_MODE,27.6
4,4,EXT_SOURCE_3,429.6
5,5,FLAG_EMP_PHONE,0.4
6,6,index,23.2
7,7,FLAG_DOCUMENT_6,0.6
8,8,BASEMENTAREA_AVG,32.2
9,9,BASEMENTAREA_MEDI,27.4


## `spec_feature_importances_ohe.csv`

**Что это за датасет:** таблица важностей признаков для `feature_matrix_spec.csv` после one-hot encoding.

**За что отвечает:** помогает понять, какие признаки из compact/spec-набора наиболее полезны. Именно этот файл мы используем в коде, чтобы выбирать top-N признаков и не загружать все 885 колонок каждый раз.

**Как читать колонки:** `Unnamed: 0` - порядковый индекс строки, `feature` - название признака, `importance` - важность признака. Если `importance = 0`, признак почти не помогал модели в расчётах автора.

**Для чего используем:** для нашего CP1 это основной файл отбора признаков: функция `load_top_features()` читает его, сортирует признаки по `importance` и выбирает самые важные колонки из `feature_matrix_spec.csv`.

In [12]:
show_head("spec_feature_importances_ohe.csv")

spec_feature_importances_ohe.csv: первые 10 строк, показано 3 колонок


,Unnamed: 0,feature,importance
0,0,FLAG_EMP_PHONE,1.4
1,1,DEF_60_CNT_SOCIAL_CIRCLE,22.8
2,2,LIVE_REGION_NOT_WORK_REGION,0.6
3,3,YEARS_BEGINEXPLUATATION_AVG,19.8
4,4,APARTMENTS_MODE,30.0
5,5,FLAG_MOBIL,0.0
6,6,FLOORSMIN_MEDI,5.2
7,7,BASEMENTAREA_AVG,21.8
8,8,LANDAREA_MODE,27.2
9,9,FLAG_DOCUMENT_21,0.0
